### PR TITLE
Set NextContinuationToken in ListObjectsV2 response for gateway

### LINKED ISSUE #4900: [gateway/s3] list object v2 does not work properly
Test doing recursive list object v2 for 1050 objects, fails.
```
java.lang.Exception: [FAILED] Test: recursive: listObject_test5(), number of items, expected: 1050, found: 21248000
        at FunctionalTest.listObject_test5(FunctionalTest.java:676)
        at FunctionalTest.runTests(FunctionalTest.java:1624)
        at FunctionalTest.main(FunctionalTest.java:1717)
```
## Steps to Reproduce (for bugs)

1.
2.
3.
4.

## Your Environment

* Version used (`minio version`):
* Environment name and version (e.g. nginx 1.9.1):
* Server type and version:
* Operating System and version (`uname -a`):
* Link to your project:

### PR DESCRIPTION
**Pull request by krishnasrinivas**
fixes #4900

## Description

NextContinuationToken response was not being set.

## How Has This Been Tested?

writing sample test program (covered by java functional tests)

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by balamurugana — APPROVED** [review]
LGTM

File: cmd/gateway-handlers.go [hunk 1]
```diff
@@ -814,8 +814,7 @@ func (api gatewayAPIHandlers) ListObjectsV2Handler(w http.ResponseWriter, r *htt
 		return
 	}
 
-	response := generateListObjectsV2Response(bucket, prefix, token, listObjectsV2Info.ContinuationToken, startAfter, delimiter, fetchOwner, listObjectsV2Info.IsTruncated, maxKeys, listObjectsV2Info.Objects, listObjectsV2Info.Prefixes)
-
+	response := generateListObjectsV2Response(bucket, prefix, token, listObjectsV2Info.NextContinuationToken, startAfter, delimiter, fetchOwner, listObjectsV2Info.IsTruncated, maxKeys, listObjectsV2Info.Objects, listObjectsV2Info.Prefixes)
```
Comment (harshavardhana) [inline cmd/gateway-handlers.go:817]: Uhoh!

**Review by harshavardhana — APPROVED** [review]

**Comment by deekoder** [comment]
waiting for @balamurugana change #4901

**Comment by balamurugana** [comment]
@krishnasrinivas Please rebase this PR

**Comment by balamurugana** [comment]
After listing 1000 objects, it fails

error occured
ErrorResponse(code=NotImplemented, message=A header you provided implies functionality that is not implemented, bucketName=null, objectName=null, resource=/minio-java-test-11lqkou, requestId=3L137, hostId=3L137)
request={method=GET, url=http://127.0.0.1:9000/minio-java-test-11lqkou?list-type=2&continuation-token=2%2188%21MDAwMDIyIW1pbmlvLWphdmEtdGVzdC1yZGJua3UhMDAwMDI4ITk5OTktMTItMzFUMjM6NTk6NTkuOTk5OTk5OVoh&prefix=minio, headers=Host: 127.0.0.1:9000
User-Agent: Minio (amd64; amd64) minio-java/dev
x-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
x-amz-date: 20170912T231946Z
Authorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20170912/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*
}
response={code=501, headers=Accept-Ranges: bytes
Content-Type: application/xml
Server: Minio/DEVELOPMENT.2017-09-12T23-06-56Z (linux; amd64)
Vary: Origin
X-Amz-Bucket-Region: us-east-1
X-Amz-Request-Id: 14E3C08708C499D2
Date: Tue, 12 Sep 2017 23:19:46 GMT
Transfer-Encoding: chunked
}

	at io.minio.MinioClient.execute(MinioClient.java:1192)
	at io.minio.MinioClient.executeGet(MinioClient.java:1285)
	at io.minio.MinioClient.listObjectsV2(MinioClient.java:2702)
	at io.minio.MinioClient.access$100(MinioClient.java:158)
	at io.minio.MinioClient$5$1.populate(MinioClient.java:2576)
	at io.minio.MinioClient$5$1.hasNext(MinioClient.java:2604)
	at FunctionalTest.listObject_test5(FunctionalTest.java:1113)
	at FunctionalTest.runTests(FunctionalTest.java:2366)
	at FunctionalTest.main(FunctionalTest.java:2474)

**Pull request merged**